File: 0167_Two_Sum_II_-_Input_array_is_sorted/main.cpp
class Solution {
  public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        int i = 0, j = numbers.size() - 1;
        while (i < j) {
            int n = numbers[i] + numbers[j];
            if (n > target) {
                j--;
            } else if (n < target) {
                i++;
            } else {
                return {i + 1, j + 1};
            }
        }
        return {-1, -1};
    }
};

File: 0167_Two_Sum_II_-_Input_array_is_sorted/golang/main.go
package solution

func twoSum(numbers []int, target int) []int {
	if len(numbers) == 0 {
		return nil
	}
	n := len(numbers)
	i, j := 0, n-1
	for i < j {
		sum := numbers[i] + numbers[j]
		if sum == target {
			return []int{i + 1, j + 1}
		}
		if sum < target {
			i++
		} else {
			j--
		}
	}
	return nil
}
